### PR TITLE
Add com.ugetdm.uGet

### PR DESCRIPTION
**Pull request by faveoled**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** [https://github.com/ugetdm/uget/issues/42](https://github.com/ugetdm/uget/issues/42)
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned.
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by faveoled** [comment]
bot, build com.ugetdm.uGet

File: aria2.yaml
Comment (bbhtt) [inline aria2.yaml:1]: Add to main manifest

File: c-ares.yaml
Comment (bbhtt) [inline c-ares.yaml:1]: Add to main manifest

File: com.ugetdm.uGet.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  <url type="translate">https://ugetdm.com/translations</url>
+  <kudos>
+    <kudo>ModernToolkit</kudo>
+    <kudo>HiDpiIcon</kudo>
+    <kudo>HighContrast</kudo>
+  </kudos>
+  <screenshots>
+    <screenshot type="default">
+      <caption>Main window</caption>
+      <image>https://github.com/scx/uget-flatpak/blob/master/uget-flatpak.png?raw=true</image>
```
Comment (bbhtt) [inline com.ugetdm.uGet.appdata.xml:49]: Screenshot should be from upstream. If they don't have one submit it.

File: com.ugetdm.uGet.appdata.xml
Comment (bbhtt) [inline com.ugetdm.uGet.appdata.xml:1]: Can you submit this upstream?

File: com.ugetdm.uGet.desktop
Comment (bbhtt) [inline com.ugetdm.uGet.desktop:1]: Can you submit this upstream?

File: com.ugetdm.uGet.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  - --socket=pulseaudio
+  # Filesystem access
+  - --filesystem=xdg-download
+  # Network access
+  - --share=network
+  # Dconf access
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --talk-name=ca.desrt.dconf
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
```
Comment (bbhtt) [inline com.ugetdm.uGet.yaml:22]: Why?

File: libssh2.yaml
Comment (bbhtt) [inline libssh2.yaml:1]: Add to main manifest

File: libssh2.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: libssh2
+buildsystem: cmake-ninja
+builddir: true
+config-opts:
+  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  - -DCMAKE_INSTALL_LIBDIR:PATH=/app/lib
```
Comment (bbhtt) [inline libssh2.yaml:6]:
```suggestion
```
Flatpak builder sets this

File: uGet.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    sha256: 11356e4242151b9014fa6209c1f0360b699b72ef8ab47dbeb81cc23be7db9049
+  - type: patch
+    path: gcc10.patch
+  - type: file
+    path: com.ugetdm.uGet.desktop
+  - type: file
+    path: com.ugetdm.uGet.appdata.xml
+
+post-install:
+  - icon_in="pixmaps/icons/hicolor/scalable/apps/uget-icon.svg";
```
Comment (bbhtt) [inline uGet.yaml:14]: Just do the svg, no need to install icons of every size unless specifically required.

**Comment by faveoled** [comment]
Build files based on https://github.com/scx/uget-flatpak

File: uGet.yaml (hunk 4 above)
Comment (faveoled) [reply uGet.yaml:14]: There was a strange Flatpak error which said something along the lines of uknown image type

File: com.ugetdm.uGet.yaml (hunk 2 above)
Comment (faveoled) [reply com.ugetdm.uGet.yaml:22]: Required for DConf?

Comment (bbhtt) [reply com.ugetdm.uGet.yaml:22]: Did you see the first line in https://docs.flatpak.org/en/latest/sandbox-permissions.html#dconf-access ?

File: uGet.yaml (hunk 4 above)
Comment (bbhtt) [reply uGet.yaml:14]: then just do a 512px

File: com.ugetdm.uGet.yaml (hunk 2 above)
Comment (faveoled) [reply com.ugetdm.uGet.yaml:22]: I didn't, but I'd need help with the new conf lines and testing

Comment (bbhtt) [reply com.ugetdm.uGet.yaml:22]: You can just remove those 4 lines and everything should work (check).

The rest of the doc (migration) is for applications that existed prior to that runtime/glib/portal.

File: com.ugetdm.uGet.appdata.xml
Comment (faveoled) [reply com.ugetdm.uGet.appdata.xml:1]: No, no idea how to use sourceforge

File: com.ugetdm.uGet.desktop
Comment (faveoled) [reply com.ugetdm.uGet.desktop:1]: No, no idea how to use sourceforge

File: com.ugetdm.uGet.yaml (hunk 2 above)
Comment (faveoled) [reply com.ugetdm.uGet.yaml]: Thanks, removed

File: uGet.yaml (hunk 4 above)
Comment (faveoled) [reply uGet.yaml:14]: Done

File: com.ugetdm.uGet.appdata.xml (hunk 1 above)
Comment (faveoled) [reply com.ugetdm.uGet.appdata.xml:49]: I'd say they're low quality - they show too much of a user desktop

File: com.ugetdm.uGet.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    </screenshot>
+  </screenshots>
+  <provides>
+    <binary>uget-gtk</binary>
+  </provides>
+  <launchable type="desktop-id">com.ugetdm.uGet.desktop</launchable>
+  <content_rating type="oars-1.0" />
```
Comment (bbhtt) [inline com.ugetdm.uGet.appdata.xml:56]:
```suggestion
  <content_rating type="oars-1.1" />
```

**Comment by faveoled** [comment]
bot, build com.ugetdm.uGet

**Comment by bbhtt** [comment]
bot, build com.ugetdm.uGet

File: com.ugetdm.uGet.yaml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        url: https://github.com/aria2/aria2/releases/download/release-1.37.0/aria2-1.37.0.tar.xz
+        sha256: 60a420ad7085eb616cb6e2bdf0a7206d68ff3d37fb5a956dc44242eb2f79b66b
+        x-checker-data:
+          type: anitya
+          project-id: 109
+          url-template: https://github.com/aria2/aria2/releases/download/release-$version/aria2-$version.tar.xz  
+    post-install:
+      - install -p -D -m 0644 "COPYING" LICENSE* -t "${FLATPAK_DEST}/share/licenses/aria2/";
+
+  - name: c-ares
```
Comment (bbhtt) [inline com.ugetdm.uGet.yaml:51]: c-ares and libssh2 must be before aria2

File: com.ugetdm.uGet.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.ugetdm.uGet</id>
+  <metadata_license>CC0</metadata_license>
+  <project_license>LGPL-2.0</project_license>
+  <name>uGet</name>
+  <summary>Download manager using GTK+ and libcurl</summary>
```
Comment (hfiguiere) [inline com.ugetdm.uGet.appdata.xml:7]:
```suggestion
  <summary>Download manager</summary>
```
https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/quality-guidelines

File: com.ugetdm.uGet.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  <description>
+    <p>uGet is a lightweight yet powerful Open Source download manager for GNU/Linux developed with GTK+.</p>
```
Comment (hfiguiere) [inline com.ugetdm.uGet.appdata.xml:9]:
```suggestion
    <p>uGet is a lightweight yet powerful Open Source download manager.</p>
```

File: com.ugetdm.uGet.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    </ul>
+  </description>
+  <developer_name>uGet Team</developer_name>
+  <url type="bugtracker">https://sourceforge.net/p/urlget/bugs</url>
+  <url type="donation">https://ugetdm.com/donate</url>
+  <url type="faq">https://ugetdm.com/qa/faq</url>
+  <url type="help">https://ugetdm.com/help</url>
+  <url type="homepage">https://ugetdm.com</url>
+  <url type="translate">https://ugetdm.com/translations</url>
+  <url type="vcs-browser">https://sourceforge.net/p/urlget/uget2/ci/master/tree/</url>
```
Comment (hfiguiere) [inline com.ugetdm.uGet.appdata.xml:40]:
```suggestion
  <url type="vcs-browser">https://sourceforge.net/p/urlget/uget2/</url>
```

File: com.ugetdm.uGet.desktop
Comment (hfiguiere) [inline com.ugetdm.uGet.desktop:1]: There is this upstream:

https://sourceforge.net/p/urlget/uget2/ci/master/tree/uget-gtk.desktop

No need for the file here.

File: com.ugetdm.uGet.yaml [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        path: gcc10.patch
+      - type: file
+        path: com.ugetdm.uGet.desktop
+      - type: file
+        path: com.ugetdm.uGet.appdata.xml
+    
+    post-install:
+      - icon_in="pixmaps/icons/hicolor/scalable/apps/uget-icon.svg";
+        icon_out="uget-icon.png";
+        rsvg-convert "${icon_in}" -w "512" -h "512" -a -f png -o "uget-icon.png";
```
Comment (hfiguiere) [inline com.ugetdm.uGet.yaml:107]: if a SVG exist, you don't need to do anything else. Don't mae things more complicated than they should

(ie drop this)

File: com.ugetdm.uGet.yaml [hunk 11]
```diff
@@ -0,0 +1,119 @@
+app-id: com.ugetdm.uGet
+runtime: org.gnome.Platform
```
Comment (hfiguiere) [inline com.ugetdm.uGet.yaml:2]: does this really need the GNOME runtime?

File: com.ugetdm.uGet.yaml [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+runtime-version: '45'
+sdk: org.gnome.Sdk
+command: uget-gtk
+rename-icon: uget-icon
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline com.ugetdm.uGet.yaml:10]: does it really support wayland?

File: com.ugetdm.uGet.yaml [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+  # Network access
+  - --share=network
+  # Notifications
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+  # OpenGL access
+  - --device=dri
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+  - name: aria2
```
Comment (hfiguiere) [inline com.ugetdm.uGet.yaml:25]: I don't see where aria2 is required.

File: com.ugetdm.uGet.yaml [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+        commands:
+          - cp -p /usr/share/automake-*/config.{sub,guess} .;
+          - autoreconf -vfi;
+    post-install:
+      - install -p -D -m 0644 "LICENSE.md" -t "${FLATPAK_DEST}/share/licenses/c-ares/";
+    cleanup:
+      - /include
+      - /share
+
+  - name: libssh2
```
Comment (hfiguiere) [inline com.ugetdm.uGet.yaml:73]: I don't see where libssh2 is needed.

File: com.ugetdm.uGet.appdata.xml
Comment (hfiguiere) [reply com.ugetdm.uGet.appdata.xml:1]: this is how one learn.

File: com.ugetdm.uGet.yaml (hunk 6 above)
Comment (hfiguiere) [reply com.ugetdm.uGet.yaml:51]: that's the buile error, but as I mentionned I don't see where these dependencies are needed.

Comment (bbhtt) [reply com.ugetdm.uGet.yaml:51]: afaik aria2c is one of the alternative backends ugetdm supports

**Comment by faveoled** [comment]
bot, build com.ugetdm.uGet

File: com.ugetdm.uGet.desktop
Comment (hfiguiere) [reply com.ugetdm.uGet.desktop:1]: this comment still stands.

File: com.ugetdm.uGet.yaml (hunk 12 above)
Comment (faveoled) [reply com.ugetdm.uGet.yaml:10]: Yes, it's based on GTK3

File: com.ugetdm.uGet.appdata.xml (hunk 1 above)
Comment (faveoled) [reply com.ugetdm.uGet.appdata.xml]: Found one, fixed

File: com.ugetdm.uGet.yaml (hunk 11 above)
Comment (faveoled) [reply com.ugetdm.uGet.yaml:2]: Fails on Freedesktop
```
Package 'libnotify', required by 'virtual:world', not found
```

File: com.ugetdm.uGet.appdata.xml
Comment (faveoled) [reply com.ugetdm.uGet.appdata.xml:1]: Submitted
https://sourceforge.net/p/urlget/uget2/merge-requests/9/

File: com.ugetdm.uGet.desktop
Comment (faveoled) [reply com.ugetdm.uGet.desktop:1]: It appears they have it:
https://sourceforge.net/p/urlget/uget2/ci/master/tree/uget-gtk.desktop

File: com.ugetdm.uGet.yaml (hunk 11 above)
Comment (bbhtt) [reply com.ugetdm.uGet.yaml:2]: libnotify is fairly wasy to build and will also be in freedesktop runtime 23.08.

If that's the only thing, can you try including libnotify here and switch to freedesktop runtime?

Comment (faveoled) [reply com.ugetdm.uGet.yaml:2]: gstreamer also I think

File: com.ugetdm.uGet.yaml (hunk 12 above)
Comment (hfiguiere) [reply com.ugetdm.uGet.yaml:10]: given the upstream desktop file that seem to want to force X11, I'll ask again. Did you actually test it on Wayland (XWayland doesn't count) ?

File: com.ugetdm.uGet.yaml (hunk 11 above)
Comment (hfiguiere) [reply com.ugetdm.uGet.yaml:2]: it's also in freedesktop.

File: com.ugetdm.uGet.desktop
Comment (hfiguiere) [inline com.ugetdm.uGet.desktop:1]: you should use upstream desktop file. use `desktop-file-edit` to adjust it.

Remove this one.

**Comment by bbhtt** [comment]
1) settings > plugin >match order > aria2 -> then dowload doesn't work

2) It assumes the default download folder is `$HOME`, is there a way to point it to `$HOME/Downloads`?

3) tray has no icon: 

image

File: com.ugetdm.uGet.yaml [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+  - --socket=wayland
+  # Audio access
+  # for sound notifications
+  - --socket=pulseaudio
+  # Filesystem access
+  - --filesystem=xdg-download
+  # Network access
+  - --share=network
+  # Notifications
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline com.ugetdm.uGet.yaml:19]:
```suggestion
```

Runtime has libnotify >= 0.8.0 it is not required.

File: com.ugetdm.uGet.desktop
Comment (faveoled) [reply com.ugetdm.uGet.desktop:1]: right, it was a stale file

**Comment by faveoled** [comment]
bot, build com.ugetdm.uGet

**Comment by faveoled** [comment]
> 1. settings > plugin >match order > aria2 -> then dowload doesn't work

Same for me. Don't really know why.


> 2\. It assumes the default download folder is `$HOME`, is there a way to point it to `$HOME/Downloads`?

Patching the sources here I think
https://sourceforge.net/p/urlget/uget2/ci/master/tree/ui-gtk/UgtkApp.c#l1768




> 3\. tray has no icon:

Not sure, it works for me, plus menu

File: com.ugetdm.uGet.yaml (hunk 11 above)
Comment (faveoled) [reply com.ugetdm.uGet.yaml]: Switched to freedesktop

**Comment by bbhtt** [comment]
> Same for me. Don't really know why.

Is aria2 working at all in any combination? If it doesn't then remove it. Upstream is dead so nothing will be fixed.

> Patching the sources here I think

If you can point that to $HOME/Downloads that would be nice

> Not sure, it works for me, plus menu

The menu works for me too but not the icon.

**Comment by faveoled** [comment]
bot, build com.ugetdm.uGet

**Comment by faveoled** [comment]
bot, build com.ugetdm.uGet

**Comment by faveoled** [comment]
> * settings > plugin >match order > aria2 -> then dowload doesn't work
> 
> * It assumes the default download folder is `$HOME`, is there a way to point it to `$HOME/Downloads`?
> 
> * tray has no icon:

Should be fixed in the latest build

**Comment by barthalion** [comment]
bot, build com.ugetdm.uGet

**Pull request closed without merging**